libafl/src/inputs/multi.rs

```diff
@@ -0,0 +1,165 @@
+//! Definitions for inputs which have multiple distinct subcomponents.
+//!
+//! Unfortunately, since both [`serde::de::Deserialize`] and [`Clone`] require [`Sized`], it is not
+//! possible to dynamically define a single input with dynamic typing. As such, [`MultipartInput`]
+//! requires that each subcomponent be the same subtype.
+
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+
+use arrayvec::ArrayVec;
+use serde::{Deserialize, Serialize};
+
+use crate::inputs::Input;
+
+/// An input composed of multiple parts. Use in situations where subcomponents are not necessarily
+/// related, or represent distinct parts of the input.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MultipartInput<I> {
+    parts: Vec<I>,
```

Comment:
We could also have an input consisting of multipe different inputs at some point (but then we're basically doing grammar fuzzing?)

Reply:
Unfortunately it is really hard (read: impossible) for me to do dynamic types like this. I tried, but it just didn't work because the type constraints on mutator eventually need a concrete type. Users will simply need to wrap `MultipartInput<I1>` and `MultipartInput<I2>` or similar.

Reply:
Well we could do it the way we do Feedbacks, like, a MultiInput<I1, I2> and then a multiinput could have multiple multiinputs again (right?)

Reply:
The problem is the mutators -- it is not possible (to my knowledge) to have generic implementations for these with diverse input types. With this support as is, though, it shouldn't be too hard to implement what you're suggesting on a case-by-case basis.